source/android/adaptivecards/src/main/java/io/adaptivecards/renderer/readonly/MediaRenderer.java

```diff
@@ -0,0 +1,349 @@
+    {
+        ImageView posterView = null;
+        if(!(media.GetPoster().isEmpty()) || !(hostConfig.getMedia().getDefaultPoster().isEmpty()))
+        {
+            // Draw poster in posterLayout
+            Image poster = new Image();
+
+            if (!media.GetPoster().isEmpty()) {
+                poster.SetUrl(media.GetPoster());
+            } else {
+                poster.SetUrl(hostConfig.getMedia().getDefaultPoster());
+            }
+
+            poster.SetImageSize(ImageSize.Auto);
+            posterView = (ImageView) ImageRenderer.getInstance().render(renderedCard, context, fragmentManager, viewGroup, poster, cardActionHandler, hostConfig, containerStyle);
+
+            RelativeLayout.LayoutParams posterLayoutParams = (RelativeLayout.LayoutParams) posterView.getLayoutParams();
+            posterLayoutParams.addRule(RelativeLayout.CENTER_IN_PARENT, RelativeLayout.TRUE);
+            posterView.setLayoutParams(posterLayoutParams);
+        }
+        return posterView;
+    }
+
+    private ImageView renderPlayButton(
+            RenderedAdaptiveCard renderedCard,
+            Context context,
+            FragmentManager fragmentManager,
+            ViewGroup viewGroup,
+            ICardActionHandler cardActionHandler,
+            HostConfig hostConfig,
+            ContainerStyle containerStyle)
+    {
+        // Draw play button on top of poster (or instead of the poster if no poster defined)
+        ImageView playButtonView;
+        String playButtonUrl = hostConfig.getMedia().getPlayButton();
+        if(!playButtonUrl.isEmpty())
+        {
+            Image playButton = new Image();
+            playButton.SetUrl(playButtonUrl);
+            playButton.SetImageSize(ImageSize.Auto);
+
+            playButtonView = (ImageView) ImageRenderer.getInstance().render(renderedCard, context, fragmentManager, viewGroup, playButton, cardActionHandler, hostConfig, containerStyle);
+        }
+        else
+        {
+            Resources resources = context.getResources();
+            Bitmap playButton = BitmapFactory.decodeResource(resources, android.R.drawable.ic_media_play);
+
+            playButtonView = new ImageView(context);
+            playButtonView.setImageBitmap(playButton);
+            viewGroup.addView(playButtonView);
+        }
+
+        RelativeLayout.LayoutParams playButtonLayoutParams = (RelativeLayout.LayoutParams) playButtonView.getLayoutParams();
+        playButtonLayoutParams.addRule(RelativeLayout.CENTER_IN_PARENT, RelativeLayout.TRUE);
+        playButtonView.setLayoutParams(playButtonLayoutParams);
+
+        return playButtonView;
+    }
+
+    private VideoViewResizeable renderMediaPlayer(
+            Context context,
+            ViewGroup viewGroup,
+            Media media)
+    {
+        VideoViewResizeable videoView = new VideoViewResizeable(context);
+        videoView.setTag(media);
+
+        for(int i = 0; i < media.GetSources().size(); i++)
```

Comment:
rather than recalculating `media.GetSources().size()` through each iteration, should we instead use a local variable?